caikit_nlp/data_model/classification.py

```diff
@@ -0,0 +1,52 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""These interfaces can be promoted to caikit/caikit for wider usage
+when applicable to multiple modules
+"""
+# Standard
+from typing import List
+
+# First Party
+from caikit.core import DataObjectBase
+import caikit
+
+
+@caikit.core.dataobject(package="caikit_data_model.caikit_nlp")
+class Classification(DataObjectBase):
```

Comment:
Question: Is the reason for not using terms like: `SequenceClassification` or `TextClassification` that we may want to use same data model object for classification tasks in other domain like vision later on?

Reply:
Yes that is what I had envisioned here, that label and score are not really domain specific e.g. https://huggingface.co/docs/transformers/tasks/image_classification. We could consider if this is reason for "promotion" to `caikit/caikit` later if this is to be reused.

Reply:
sounds good